src/main/java/edu/cornell/kfs/fp/service/impl/ScheduledAccountingLineServiceImpl.java

```diff
@@ -0,0 +1,145 @@
+package edu.cornell.kfs.fp.service.impl;
+
+import static org.kuali.kfs.sys.document.validation.impl.AccountingDocumentRuleBaseConstants.ERROR_PATH.DOCUMENT_ERROR_PREFIX;
+
+import java.io.Serializable;
+import java.sql.Date;
+import java.util.Calendar;
+import java.util.TreeMap;
+
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.sys.context.SpringContext;
+import org.kuali.kfs.sys.util.KfsDateUtils;
+import org.kuali.rice.core.api.util.type.KualiDecimal;
+import org.kuali.rice.coreservice.framework.parameter.ParameterService;
+import org.kuali.rice.krad.util.GlobalVariables;
+
+import edu.cornell.kfs.fp.CuFPConstants;
+import edu.cornell.kfs.fp.businessobject.ScheduledSourceAccountingLine;
+import edu.cornell.kfs.gl.service.ScheduledAccountingLineService;
+import edu.cornell.kfs.sys.CUKFSKeyConstants;
+
+public class ScheduledAccountingLineServiceImpl implements ScheduledAccountingLineService, Serializable {
+
+    private static final long serialVersionUID = -5700446522442741269L;
+    private transient ParameterService parameterService;
+    
+    private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(ScheduledAccountingLineServiceImpl.class);
+
+    public TreeMap<Date, KualiDecimal> generateDatesAndAmounts(ScheduledSourceAccountingLine scheduledAccountingLine, int rowId) {
+        TreeMap<Date, KualiDecimal> datesAndAmounts = new TreeMap<Date, KualiDecimal>();
+        Date endDate = generateEndDate(scheduledAccountingLine);
+        scheduledAccountingLine.setEndDate(endDate);
+        KualiDecimal totalAmount = calculateTotalAmount(scheduledAccountingLine.getAmount());
+
+        Date currentStartDate = (Date) scheduledAccountingLine.getStartDate().clone();
+        int iterationCount = 0;
+
+        while (validDateForLooping(endDate, currentStartDate) && totalAmount.isGreaterThan(KualiDecimal.ZERO)) {
+            Calendar calendarCalculator = Calendar.getInstance();
+            calendarCalculator.setTime(scheduledAccountingLine.getStartDate());
+
+            calendarCalculator = incrementDateByValue(scheduledAccountingLine, calendarCalculator, iterationCount);
+
+            Date entryDate = new Date(calendarCalculator.getTimeInMillis());
+            addToDateAndAmount(datesAndAmounts, scheduledAccountingLine, entryDate, totalAmount, endDate);
+
+            currentStartDate = new Date(calendarCalculator.getTimeInMillis());
+            totalAmount = totalAmount.subtract(scheduledAccountingLine.getPartialAmount());
+
+            iterationCount++;
+        }
+
+        determineAccountingLineErrors(scheduledAccountingLine, rowId, iterationCount);
+
+        return datesAndAmounts;
+    }
+
+    private boolean validDateForLooping(Date endDate, Date currentStartDate) {
```

Comment:
I'm not sure this name is super meaningful. Maybe it is in context, but would isSameDayOrEarlier be more descriptive?